src/pycolmap/scene/image.cc

```diff
@@ -26,10 +26,15 @@ std::string PrintImage(const Image& image) {
   std::stringstream ss;
   ss << "Image(image_id="
      << (image.ImageId() != kInvalidImageId ? std::to_string(image.ImageId())
-                                            : "Invalid")
-     << ", camera_id="
-     << (image.HasCamera() ? std::to_string(image.CameraId()) : "Invalid")
-     << ", name=\"" << image.Name() << "\""
+                                            : "Invalid");
+  if (!image.HasCameraPtr()) {
```

Comment:
Assuming that the camera ptr and the id are always consistent, we would not need this branch here?

Reply:
Here I update the printing to show to the user if the optional camera ptr is set or not. Check the python example in the PR description (the printing is slightly different)